src/archemist/core/optimisation/optimisation_manager.py

```diff
@@ -6,23 +6,27 @@
 from archemist.core.state.optimisation_state import OptimizationState
 from archemist.core.persistence.object_factory import OptimizationFactory
 from archemist.core.persistence.persistence_manager import PersistenceManager
+from archemist.core.persistence.recipe_files_watchdog import RecipeFilesWatchdog
 from recipe_generator import RecipeGenerator
 from pathlib import Path
-from threading import Thread
-
+import importlib
 
 class OptimisationManager():
 
-    def __init__(self, workflow_dir: str, state) -> None:
+    def __init__(self, workflow_dir: str, recipes_dir_path: str, recipes_watchdog, state) -> None:
         self._config_file = Path.joinpath(workflow_dir, "config_files/optimization_config.yaml")
-        self._recipes_dir = Path.joinpath(workflow_dir, "recipes")
+        self._recipes_dir = recipes_dir_path
+        self._recipes_watchdog = recipes_watchdog
```

Comment:
as per my comment below, you don't need the recipe watchdog to access the recipe queue. State is adequate for that purpose